mobile-app/src/screens/Fitness-Diet/index.js

```diff
@@ -71,6 +75,62 @@ const FitnessDiet = ({ navigation }) => {
     }
   }, []);
 
+  const createDietButtonResponse = async (input) => {
+    try {
+      var token = await getAccessToken();
+      const response = await DietButtonAPI.createDietButtonResponse(token, input);
+
+      await getDietButtonResponse();
+    }
+    catch (e){
+      setIsLoading(false);
+      if (Platform.OS === "ios") {
+        Toast.show("Something went wrong. Please try again.", {
+          ...styles.errorToast,
+          duration: Toast.durations.LONG,
+          position: Toast.positions.BOTTOM,
+        });
+      } else {
+        Toast.show("Something went wrong. Please try again.", {
+          ...styles.errorToast,
+          duration: Toast.durations.LONG,
+          position: Toast.positions.TOP,
+        });
+      }
+    }
+  };
+
+  const getDietButtonResponse = async () => {
+    try {
+      token = await getAccessToken();
+      const responses = DietButtonAPI.getDietButtonResponse(token);
+      setButtonResponses(responses);
```

Comment:
You don't want to use a state variable here. The response isn't used to update any actual UI and is passed into the Toast _after_ the you make a call to DietButtonAPI